sstables/sstable_set.cc

```diff
@@ -1380,7 +1381,7 @@ sstable_set::make_local_shard_sstable_reader(
         }
         return reader;
     };
-    if (_impl->size() == 1) [[unlikely]] {
+    if (_impl->size() == 1 && !histogram) [[unlikely]] {
```

Comment:
I don't think `[[unlikely]]` is justified here. This function is used by user reads too, which will never pass `histogram` here.